pandas/_libs/tslibs/conversion.pyx

```diff
@@ -355,26 +365,16 @@ cdef inline void _localize_tso(_TSObject obj, object tz):
         # static/pytz/dateutil specific code
         if is_fixed_offset(tz):
             # statictzinfo
-            if len(deltas) > 0 and obj.value != NPY_NAT:
```

Comment:
you are eliminating the not len(deltas) case

Reply:
Right.  `get_dst_info` never returns an empty array in the `is_fixed_offset` case.

Reply:
More specifically: it always returns a length-1 array.

Reply:
hmm can u add an assert to that effect then

Reply:
OK.  As long as we're asserting things, I think it is the case that in _localize_tso the input `obj` will always have `obj.tzinfo is None`; can we make that official?  Even in the docstring would be a win.

Reply:
asserts in internal code are a good thing
happy to add; they are just laying out the assumptions explicitly 
don’t go crazy but a reader of the code will appreciate

Reply:
Great.  That'll close out one of the tslibs-todo questions.  The last one since we're on a role: at the moment all calls to tz_convert or tz_convert_single have at least one of tz1 or tz2 being utc.  Will that always be the case?  If so that'll close [ill have to look it up].

Reply:
iiuc i think so
we never convert from a time zone to another one directly
always pass thru utc

Reply:
Excellent.  I'll ping you on the appropriate bug-issue to close that out; don't want to derail this.